### PR TITLE
Introduce `http` config settings in Azure storage

### PR DESCRIPTION
**Pull request by siggy**
Cortex v1.11.0 included thanos-io/thanos#3970, which added configuration
options to Azure's http client and transport, replacing usage of
`http.DefaultClient`. Unfortunately since Cortex was not setting this
config, Cortex implicitly switched from `http.DefaultClient` to all
empty values (e.g. `MaxIdleConns: 0` rather than 100).

Introduce `http` config settings to Azure storage. This motivated moving
`s3.HTTPConfig` into a new `pkg/storage/bucket/config` package, to allow
`azure` and `s3` to share it.

Also update the instructions for running the website to include
installing `embedmd`.

Signed-off-by: Andrew Seigner <andrew@sig.gy>

**What this PR does**:

Introduces a new `http` config for Azure storage, similar to `s3`.

**Which issue(s) this PR fixes**:

Relates to:
- thanos-io/thanos#3970
- thanos-io/thanos#4928

With these new http defaults, we observed a 10%~40% memory reduction in alertmanager, compactor, and ruler, and more stable memory usage:

cortex http config

Note the above measurements assume thanos-io/thanos#4928 is merged into Cortex. That change provides a much larger memory reduction, and without it, the memory savings from this PR is negligible.

**Checklist**
- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,14 +3,15 @@
 ## master / unreleased
 
 * [CHANGE] Changed default for `-ingester.min-ready-duration` from 1 minute to 15 seconds. #4539
-* [ENHANCEMENT] Ruler: Add `ruler.disable-rule-group-label` to disable the `rule_group` label on exported metrics. #4571
+* [CHANGE] query-frontend: Do not print anything in the logs of `query-frontend` if a in-progress query has been canceled (context canceled). #4562
```
Comment (bboreham) [inline CHANGELOG.md:6]: Unrelated change - please move to a different PR.

File: pkg/storage/bucket/config/http.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package config
```
Comment (bboreham) [inline pkg/storage/bucket/config/http.go:1]: Small thing, but I got a little confused over the naming of this package.
It doesn't contain all bucket config.  I think it exists to avoid a circular dependency?
Given what it does, I might call the package `http` and the struct `Config`.

**Review by bboreham — APPROVED** [review]
Thanks!  Couple of thoughts below.

File: CHANGELOG.md (hunk 1 above)
Comment (siggy) [reply CHANGELOG.md:6]: Posted #4584

File: pkg/storage/bucket/config/http.go (hunk 2 above)
Comment (siggy) [reply pkg/storage/bucket/config/http.go]: Yup it was to avoid the circular dependency. Alternatively we could make `azure` depend on `s3.HTTPConfig`, but that felt worse, also `s3.HTTPConfig` has a `Transport` that would be unused by `azure`.

I've renamed `config.HTTP` -> `http.Config` in the latest commit.

**Review by bboreham — APPROVED** [review]

**Review by alvinlin123 — APPROVED** [review]
Looks good. Thanks!
